components/org.wso2.carbon.identity.user.ekyc/src/main/java/org/wso2/carbon/identity/user/ekyc/idv/IDVServiceImpl.java

```
@@ -0,0 +1,166 @@
+import org.apache.http.client.HttpClient;
+import org.apache.http.client.methods.HttpPost;
+import org.apache.http.conn.ssl.AllowAllHostnameVerifier;
+import org.apache.http.conn.ssl.SSLContextBuilder;
+import org.apache.http.conn.ssl.TrustStrategy;
+import org.apache.http.entity.ContentType;
+import org.apache.http.entity.StringEntity;
+import org.apache.http.impl.client.CloseableHttpClient;
+import org.apache.http.impl.client.HttpClients;
+import org.apache.http.util.EntityUtils;
+import org.wso2.carbon.identity.configuration.mgt.core.ConfigurationManager;
+import org.wso2.carbon.identity.configuration.mgt.core.exception.ConfigurationManagementException;
+import org.wso2.carbon.identity.configuration.mgt.core.model.Resource;
+import org.wso2.carbon.identity.user.ekyc.dto.*;
+import org.wso2.carbon.identity.user.ekyc.exception.IDVException;
+import org.wso2.carbon.identity.user.ekyc.util.IDVConstants;
+
+import java.io.IOException;
+import java.io.UnsupportedEncodingException;
+import java.security.KeyManagementException;
+import java.security.KeyStoreException;
+import java.security.NoSuchAlgorithmException;
+import java.security.cert.CertificateException;
+import java.security.cert.X509Certificate;
+import java.util.List;
+
+import static org.wso2.carbon.identity.user.ekyc.util.EKYCConfigurationMapper.buildEKYCConfigurationFromResource;
+import static org.wso2.carbon.identity.user.ekyc.util.IDVConstants.EKYC_RESOURCE_TYPE;
+import static org.wso2.carbon.identity.user.ekyc.util.IDVConstants.RESOURCE_NAME;
+
+public class IDVServiceImpl implements IDVService {
+
+    private HttpClient httpClient;
+    private ConfigurationManager configurationManager;
+
+    private static final Log log = LogFactory.getLog(IDVServiceImpl.class);
+
+    public IDVServiceImpl(ConfigurationManager configurationManager) {
+        this.configurationManager = configurationManager;
+        try {
+            httpClient = newClient();
+        } catch (Exception e) {
```

Can we catch the specific exception here? Also if the httpClient creation fails shall we throw the exception instead of simply logging it. As this will cause nullpointers when the httpClient will be used.

---

Done